scala-package/examples/src/main/java/org/apache/mxnetexamples/javaapi/infer/bert/BertUtil.java

```diff
@@ -0,0 +1,121 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.mxnetexamples.javaapi.infer.bert;
+
+import java.io.FileReader;
+import java.util.*;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+public class BertUtil {
+
+    private Map<String, Integer> token2idx;
+    private List<String> idx2token;
+
+    /**
+     * Parse the Vocabulary to JSON files
+     * [PAD], [CLS], [SEP], [MASK], [UNK] are reserved token
+     * @param jsonFile the filePath of the vocab.json
+     * @throws Exception
+     */
+    void parseJSON(String jsonFile) throws Exception {
+        Gson gson = new Gson();
+        token2idx = new HashMap<>();
+        idx2token = new LinkedList<>();
+        JsonObject jsonObject = gson.fromJson(new FileReader(jsonFile), JsonObject.class);
+        JsonArray arr = jsonObject.getAsJsonArray("idx_to_token");
+        for (JsonElement element : arr) {
+            idx2token.add(element.getAsString());
+        }
+        JsonObject preMap = jsonObject.getAsJsonObject("token_to_idx");
+        for (String key : preMap.keySet()) {
+            token2idx.put(key, preMap.get(key).getAsInt());
+        }
+    }
+
+    /**
+     * Tokenize the input, split all kinds of spaces and
+     * saparate the end of sentence symbol: . , ? !
+     * @param input The input String
+     * @return List of tokens
+     */
+    List<String> tokenizer(String input) {
+        String[] step1 = input.split("[\n\r\t ]+");
+        List<String> finalResult = new LinkedList<>();
+        for (String item : step1) {
+            if (item.length() != 0) {
+                if ((item + "a").split("[.,?!]+").length > 1) {
+                    finalResult.add(item.substring(0, item.length() - 1));
+                    finalResult.add(item.substring(item.length() -1));
+                } else {
+                    finalResult.add(item);
+                }
+            }
+        }
+        return finalResult;
+    }
+
+    /**
+     * Pad the tokens to the required length
+     * @param tokens input tokens
+     * @param padItem things to pad at the end
+     * @param num total length after padding
+     * @return List of padded tokens
+     */
+    <E> List<E> pad(List<E> tokens, E padItem, int num) {
+        if (tokens.size() >= num) return tokens;
+        List<E> padded = new LinkedList<>(tokens);
+        for (int i = 0; i < num - tokens.size(); i++) {
+            padded.add(padItem);
+        }
+        return padded;
+    }
+
+    /**
+     * Convert tokens to indexes
+     * @param tokens input tokens
+     * @return List of indexes
+     */
+    List<Integer> token2idx(List<String> tokens) {
+        List<Integer> indexes = new ArrayList<>();
+        for (String token : tokens) {
+            if (token2idx.containsKey(token)) {
```

Comment:
This can be written succinctly as : 
```
int idx = token2idx.getOrDefault(token, token2idx.get("[UNK]");
indexes.add(idx);
```

Reply:
> This can be written succinctly as :
> 
> ```
> int idx = token2idx.getOrDefault(token, token2idx.get("[UNK]");
> indexes.add(idx);
> ```

This is a Java8 feature, I tried.

Reply:
Oh right. 
I forgot we support Java 7+.